Load particles according to edm4hep (#36)
key4hep/dmx: model/MODEL.md

# Keeping dmx up to date

To keep dmx up to date with the event data model [edm4hep](https://github.com/key4hep/EDM4hep), we need a way to get the latest information for each datatype and component. Also, we need to take into account that dmx is deployed on github pages as a static website, so we need a way to update this info without having a server running.

## How to get the latest information?

In the root directory, there are two commands in `package-json` that can be used to get the latest information from the edm4hep repository. The first command

```bash
npm run download
```

will download the latest information from the edm4hep, by using the file called `edm4hep.yaml` under `model` directory, containg up to date information about `edm4hep`. The second command

> 

Reviewer: typo
```suggestion
will download the latest information from the edm4hep, by using the file called `edm4hep.yaml` under `model` directory, containing up to date information about `edm4hep`. The second command
```

```bash
npm run build
```

will generate a file called `datatypes.js` under `output` directory, where all the information is parsed and stored in a format that can be used by the dmx application.

## All at once

You can also run

```bash
npm run update
```

> to run both processes at once and inmediatly update the dmx application.

Reviewer 2: ```suggestion
to run both processes at once and immediatly update the dmx application.
```

> to run both processes at once and immediatly update the dmx application.

Reviewer: typo
```suggestion
to run both processes at once and immediately update the dmx application.
```

## Further improvements

> -   We could trigger the update process automatically when a new release of edm4hep is published, but for now, we will keep it simple and run the update process manually.

Reviewer 2: I agree with keeping it manual for the moment. Especially because we might need manual intervention in any case. What we could do in a workflow would be to simply run the extraction and compare it o the current `datatypes.js` to see if there are any differences.

Reviewer 3: This is also comment for the next PRs: What about checking just schema version? @tmadlener, will this be reliable way to detect changes after EDM4hep 1.0?

Reviewer 2: Yes, In principle EDM4hep versions can change without schema version changes, but there will be no schema version change without an EDM4hep version change.

> -   To maitain backward compatibility, we could add a versioning system so we can later load different versions of the edm4hep data model into dmx, however we currrently load the latest version.

Reviewer: typos
```suggestion
-   To maintain backward compatibility, we could add a versioning system so we can later load different versions of the edm4hep data model into dmx, however we currently load the latest version.
```
